conda/common/compat.py

```diff
@@ -13,13 +13,15 @@
 import sys
 from tempfile import mkdtemp
 import warnings as _warnings
+import platform
 
 on_win = bool(sys.platform == "win32")
 on_mac = bool(sys.platform == "darwin")
 on_linux = bool(sys.platform == "linux")
 
 PY2 = sys.version_info[0] == 2
 PY3 = sys.version_info[0] == 3
+PYPY = platform.python_implementation() == "PyPy"
```

Comment:
It is "cheaper" to do `sys.implementation.name == 'pypy'`, since importing platform makes system calls